core/iwasm/common/wasm_runtime_common.c

```diff
@@ -2482,6 +2482,50 @@ wasm_runtime_get_custom_data(WASMModuleInstanceCommon *module_inst_comm)
     return module_inst->custom_data;
 }
 
+#if WASM_CONFIGUABLE_BOUNDS_CHECKS != 0
+void
+wasm_runtime_set_bounds_checks(WASMModuleInstanceCommon *module_inst,
+                               bool enable)
+{
+#if WASM_ENABLE_INTERP != 0
+    if (module_inst->module_type == Wasm_Module_Bytecode) {
+        ((WASMModuleInstanceExtra *)((WASMModuleInstance *)module_inst)->e)
+            ->disable_bounds_checks = enable ? false : true;
```

Comment:
An issue is that whether we should consider the `OS_ENABLE_HW_BOUND_CHECK`: when it is defined and hw bound check is enabled, shall we always set disable_bounds_checks true here?

Reply:
I think it's ok, maybe this work with OS_ENABLE_HW_BOUND_CHECK will make ther performance better.